src/lib/paginator/paginator.spec.ts

```diff
@@ -0,0 +1,303 @@
+import {MdPaginator, PageEvent} from './paginator';
+import {Component, ElementRef, ViewChild} from '@angular/core';
+import {MdCommonModule} from '../core';
+import {MdSelectModule} from '../select/index';
+import {MdTooltipModule} from '../tooltip/index';
+import {MdButtonModule} from '../button/index';
+import {MdPaginatorIntl} from './paginator-intl';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {customMatchers} from '../core/testing/jasmine-matchers';
+import {dispatchMouseEvent} from '../core/testing/dispatch-events';
+
+
+describe('MdPaginator', () => {
+  let fixture: ComponentFixture<MdPaginatorApp>;
+  let component: MdPaginatorApp;
+  let paginator: MdPaginator;
+
+  beforeEach(async(() => {
+    jasmine.addMatchers(customMatchers);
+
+    TestBed.configureTestingModule({
+      imports: [
+        MdButtonModule,
+        MdCommonModule,
+        MdSelectModule,
+        MdTooltipModule,
+        MdPaginatorModule,
+        NoopAnimationsModule,
+      ],
+      declarations: [
+        MdPaginatorApp,
+        MdPaginatorWithoutPageSizeOrOptionsApp,
+        MdPaginatorWithoutPageSizeApp,
+        MdPaginatorWithoutOptionsApp,
+      ],
+      providers: [MdPaginatorIntl]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(MdPaginatorApp);
+    component = fixture.componentInstance;
+    paginator = component.mdPaginator;
+
+    fixture.detectChanges();
+  }));
+
+  describe('with the default internationalization provider', () => {
+    it('should show the right range text', () => {
+      const rangeElement = fixture.nativeElement.querySelector('.mat-paginator-range-label');
+
+      // View second page of list of 100, each page contains 10 items.
+      component.length = 100;
+      component.pageSize = 10;
+      component.pageIndex = 1;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('11 - 20 of 100');
+
+      // View third page of list of 200, each page contains 20 items.
+      component.length = 200;
+      component.pageSize = 20;
+      component.pageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('41 - 60 of 200');
+
+      // View first page of list of 0, each page contains 5 items.
+      component.length = 0;
+      component.pageSize = 5;
+      component.pageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('0 of 0');
+
+      // View third page of list of 12, each page contains 5 items.
+      component.length = 12;
+      component.pageSize = 5;
+      component.pageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('11 - 12 of 12');
+
+      // View third page of list of 10, each page contains 5 items.
+      component.length = 10;
+      component.pageSize = 5;
+      component.pageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('11 - 15 of 10');
```

Comment:
Definitely is wrong =D but its wrong from the developer since they are telling us to go to a page that should not exist. Other alternative is to throw an error, but we opted to go the route of displaying what the developer told us to display (that is, items 11-15).